front/lib/api/chat.ts

```diff
@@ -104,116 +111,102 @@ export async function getChatSession(
   };
 }
 
+export async function getChatMessage(
+  sId: string
+): Promise<ChatMessageType | null> {
+  const chatMessage = await ChatMessage.findOne({
+    where: { sId },
+  });
+  if (!chatMessage) {
+    return null;
+  }
+  const retrievedDocuments = await ChatRetrievedDocument.findAll({
+    where: {
+      chatMessageId: chatMessage.id,
+    },
+    order: [["score", "DESC"]],
+  });
+
+  return {
+    role: chatMessage.role,
+    message: chatMessage.message,
+    feedback: chatMessage.feedback,
+    sId: chatMessage.sId,
+    retrievals: retrievedDocuments.map((r) => {
+      return {
+        dataSourceId: r.dataSourceId,
+        sourceUrl: r.sourceUrl,
+        documentId: r.documentId,
+        timestamp: r.timestamp,
+        tags: r.tags,
+        score: r.score,
+        chunks: [],
+      };
+    }),
+  };
+}
+
 export function newChatSessionId() {
   const uId = new_id();
   return uId.slice(0, 10);
 }
 
-export async function storeChatSession(
-  sId: string,
-  owner: WorkspaceType,
-  user: UserType,
-  title: string | null,
-  messages: ChatMessageType[]
-): Promise<ChatSessionType> {
+export async function upsertChatMessage(
+  sessionId: number,
```

Comment:
Ideally if we can pass the `session: ChatSessionType` it's marginally better as it leaks less the postgres abstractions. But not super important.

Reply:
I still thing the interface would be nicer if we used `session: ChatSessionType` here?